File: digae_backend/src/main/java/com/digae/sga/dto/response/SupervisionResponseDTO.java
package com.digae.sga.dto.response;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SupervisionResponseDTO {

    private Long id;
    private LocalDate fechaInspeccion;
    private String areaInspeccionada;
    private java.util.List<com.digae.sga.dto.request.ItemSupervisionDTO> items;
    private BigDecimal calificacionGeneral;
    private String observaciones;
    private Long inspectorId;
    private String inspectorNombreCompleto;
    private LocalDateTime creadoEn;
}

File: digae_android/app/src/main/java/com/grupo7poo2/digae/network/dto/SupervisionResponseDTO.kt
package com.grupo7poo2.digae.network.dto

import com.google.gson.annotations.SerializedName

data class SupervisionResponseDTO(
    @SerializedName("id")
    val id: Long,
    @SerializedName("fechaInspeccion")
    val fechaInspeccion: String,
    @SerializedName("areaInspeccionada")
    val areaInspeccionada: String,
    @SerializedName("calificacionGeneral")
    val calificacionGeneral: Double,
    @SerializedName("observaciones")
    val observaciones: String?,
    @SerializedName("inspectorId")
    val inspectorId: Long,
    @SerializedName("inspectorNombreCompleto")
    val inspectorNombreCompleto: String?,
    @SerializedName("creadoEn")
    val creadoEn: String?,
    @SerializedName("items")
    val items: List<ItemSupervisionDTO>?
)
